crates/viewer/re_space_view_spatial/src/visualizers/depth_images.rs

```diff
@@ -335,9 +363,47 @@ impl VisualizerSystem for DepthImageVisualizer {
     }
 }
 
+impl TypedComponentFallbackProvider<DrawOrder> for DepthImageVisualizer {
+    fn fallback_for(&self, _ctx: &QueryContext<'_>) -> DrawOrder {
+        DrawOrder::DEFAULT_DEPTH_IMAGE
+    }
+}
+
+impl TypedComponentFallbackProvider<ValueRange> for DepthImageVisualizer {
+    fn fallback_for(
+        &self,
+        ctx: &re_viewer_context::QueryContext<'_>,
+    ) -> re_types::components::ValueRange {
+        if let Some(((_time, buffer_row_id), image_buffer)) = ctx
+            .recording()
+            .latest_at_component::<ImageBuffer>(ctx.target_entity_path, ctx.query)
+        {
+            // TODO(andreas): What about overrides on the image format?
+            if let Some((_, format)) = ctx
+                .recording()
+                .latest_at_component::<ImageFormat>(ctx.target_entity_path, ctx.query)
+            {
+                let image = ImageInfo {
+                    buffer_row_id,
+                    buffer: image_buffer.0,
+                    format: format.0,
+                    kind: ImageKind::Depth,
+                };
+                let cache = ctx.viewer_ctx.cache;
+                let image_stats = cache.entry(|c: &mut ImageStatsCache| c.entry(&image));
+                let default_range =
+                    ColormapWithMappingRange::default_range_for_depth_images(&image_stats);
+                return [default_range[0] as f64, default_range[1] as f64].into();
+            }
+        }
+
+        [0.0, f64::MAX].into()
```

Comment:
Does this mean the default depth range is decided by the range of the "current" depth image, whatever that may be the first time the user starts the viewer?

That seems like a fair heuristic, except for one thing: if the time starts at zero, and the latest-at comes up short, we end up with [0, f64::MAX]. I think it would be better to instead do a latest-ever query to make the chance of this case less likely

Reply:
latest-ever would require a store subscriber for this and I dont'really want to go there. As discussed on the primary-pr-review-comment-thing-thread, the behavior is as before. so let's leave it be at least for now :)